[MLIR][NVVM] Add "blocksareclusters" kernel attribute support

rajatbajpai opened the pull request:
This change adds "nvvm.blocksareclusters" kernel attribute support in NVVM Dialect/MLIR.

rajatbajpai committed:
[MLIR] Add "blocksareclusters" kernel attribute support

This change adds "nvvm.blocksareclusters" kernel attribute support.

grypp commented on mlir/lib/Dialect/LLVMIR/IR/NVVMDialect.cpp:
```diff
@@ -1922,6 +1922,15 @@ LogicalResult NVVMDialect::verifyOperationAttribute(Operation *op,
+  if (attrName == NVVMDialect::getBlocksAreClustersAttrName()) {
+    if (!op->hasAttr(NVVMDialect::getReqntidAttrName()) ||
+        !op->hasAttr(NVVMDialect::getClusterDimAttrName()))
+      return op->emitError()
+             << "'" << attrName << "' attribute must be used along with "
+             << "'" << NVVMDialect::getReqntidAttrName() << "' and "
+             << "'" << NVVMDialect::getClusterDimAttrName() << "'";
```
this is not a strict policy, but imo it's better to have {} when we've multiline code

rajatbajpai committed:
Addressed review comments

+ Added curly braces for multi line formatted statement.
+ Added small description about "blocksareclusters."

grypp reviewed (approved):
nice thanks
